components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceTypeResourceManager.java

```diff
@@ -0,0 +1,200 @@
+                schema = SCIMResourceSchemaManager.getInstance().getResourceTypeResourceSchema();
+                scimUserObjectString = encoder.buildUserResourceTypeJsonBody();
+            } else {
+                schema =
+                        SCIMResourceSchemaManager.getInstance().getResourceTypeResourceSchemaWithoutMultiValuedSchemaExtensions();
+                scimUserObjectString = buildUserResourceTypeJsonBody();
+            }
+
+            //create a string in json format for group resource type with relevant values
+            String scimGroupObjectString = encoder.buildGroupResourceTypeJsonBody();
+            //build the user abstract scim object
+            AbstractSCIMObject userResourceTypeObject = (AbstractSCIMObject) decoder.decodeResource(
+                    scimUserObjectString, schema, new AbstractSCIMObject());
+            //add meta data
+            userResourceTypeObject = ServerSideValidator.validateResourceTypeSCIMObject(userResourceTypeObject);
+            //build the group abstract scim object
+            AbstractSCIMObject groupResourceTypeObject = (AbstractSCIMObject) decoder.decodeResource(
+                    scimGroupObjectString, schema, new AbstractSCIMObject());
+            //add meta data
+            groupResourceTypeObject = ServerSideValidator.validateResourceTypeSCIMObject(groupResourceTypeObject);
+            //build the root abstract scim object
+            AbstractSCIMObject resourceTypeObject = buildCombinedResourceType(userResourceTypeObject,
+                    groupResourceTypeObject);
+            //encode the newly created SCIM Resource Type object.
+            String encodedObject;
+            Map<String, String> responseHeaders = new HashMap<String, String>();
+
+            if (resourceTypeObject != null) {
+                //create a deep copy of the resource type object since we are going to change it.
+                AbstractSCIMObject copiedObject = (AbstractSCIMObject) CopyUtil.deepCopy(resourceTypeObject);
+                encodedObject = encoder.encodeSCIMObject(copiedObject);
+                //add location header
+                responseHeaders.put(SCIMConstants.LOCATION_HEADER, getResourceEndpointURL(
+                        SCIMConstants.RESOURCE_TYPE_ENDPOINT));
+                responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+
+            } else {
+                String error = "Newly created User resource is null.";
+                throw new InternalErrorException(error);
+            }
+            //put the uri of the resource type object in the response header parameter.
+            return new SCIMResponse(ResponseCodeConstants.CODE_OK,
+                    encodedObject, responseHeaders);
+        } catch (CharonException | BadRequestException | InternalErrorException | NotFoundException e) {
+            return encodeSCIMException(e);
+        } catch (JSONException e) {
+            return null;
```

Comment:
shall we log the exception before returning

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/352/commits/9f6d7fe7474928c000edfe7c28b6e6b84cb777c7